src/main/java/org/zalando/nakadi/service/EventPublisher.java

```diff
@@ -118,6 +119,33 @@ public EventPublishResult publish(final String events, final String eventTypeNam
         }
     }
 
+    EventPublishResult publishInternal(final String events, final String eventTypeName) {
+        Closeable publishingCloser = null;
+        final List<BatchItem> batch = BatchFactory.from(events);
+        try {
+            publishingCloser = timelineSync.workWithEventType(eventTypeName, nakadiSettings.getTimelineWaitTimeoutMs());
+            final EventType eventType = eventTypeCache.getEventType(eventTypeName);
+            partition(batch, eventType);
+            submit(batch, eventType);
```

Comment:
why have you also skipped validation and enrichment? I thought you will just add one param to `publish` method which will indicate if you need authorization or not.